### PR TITLE
added cypress tests

### PR DESCRIPTION
**Pull request by ohadsahar**

### CODE REVIEW

File: core/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -13,8 +13,13 @@ const Footer = () => {
   };
 
   return (
-    <BottomSection data-testid="footer-box" onClick={handleResetProgress}>
+    <BottomSection
+      data-testid="footer-box"
+      onClick={handleResetProgress}
+      data-cy="cy-footer"
```
Comment (amitmarko) [inline core/components/Footer/Footer.tsx]: no need `data-testid` and `data-cy` please use only `data-testid`, for example:
`cy.findByTestId('footer-box')`

**Comment by amitmarko** [comment]
@ohadsahar remove all `data-cy` and use `data-testid`

File: core/components/Footer/Footer.tsx (hunk 1 above)
Comment (ohadsahar) [reply core/components/Footer/Footer.tsx]: This function does not exists on cy

**Comment by amitmarko** [comment]
Add `cypress/videos/MainPage.cy.ts.mp4` to `gitignore`

File: core/components/Footer/Footer.tsx (hunk 1 above)
Comment (amitmarko) [reply core/components/Footer/Footer.tsx]: It should found we use it:
image

File: core/components/TasksList/TasksList.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ interface TaskListProps {
   taskPendingCount?: number;
   editTaskId: string;
   listName: TaskStatusType;
+  testId?: string;
```
Comment (ssharabiappsforce) [inline core/components/TasksList/TasksList.tsx:24]:
```suggestion
  'data-testid'?: string;
```

File: cypress/e2e/integration/MainPage.cy.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import { DateProps } from "interfaces/date_props.interface";
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts:1]: Rename the file name to .spec.ts

File: cypress/e2e/integration/MainPage.cy.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    const dateRepeatableClass =
+      '[data-testid="header-box"]> [data-testid="header-date-wrapper"]';
+    const today = new Date(Date.now());
+    const convertedDate: DateProps = {
+      day: today.getDate(),
+      month: today.toLocaleString("default", { month: "long" }),
+      year: today.getFullYear(),
+      currentDay: today.toLocaleString("default", { weekday: "long" }),
+    };
+    cy.get(
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts:23]: Use in findByTestid of jest instead All places U need.

File: cypress/e2e/integration/MainPage.cy.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      `${dateRepeatableClass} > [data-testid="header-day-wrapper"] >[data-testid="header-day"]`
+    ).should("have.text", convertedDate.day);
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts:25]: If the text isn't related to the flow behavior, move the test to unit test in server env. instead.

File: shared/typography/app_title.tsx [hunk 6]
```diff
@@ -7,18 +7,20 @@ export interface AppTitleProps {
   color?: string;
   fontSize?: string;
   role?: string;
+  testId?: string;
```
Comment (ssharabiappsforce) [inline shared/typography/app_title.tsx:10]:
```suggestion
  'data-testid'?: string;
```

File: cypress/support/commands.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
Comment (ssharabiappsforce) [inline cypress/support/commands.ts:1]: For using the jest methods pls add `import '@testing-library/cypress/add-commands';` to this file

File: cypress/e2e/integration/MainPage.cy.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+      .children()
+      .should("have.length", 1);
+    cy.get(`[data-testid="finished-tasks-list"]> ${tasksRepeatableClass}`)
+      .children()
+      .should("have.length", 0);
+    cy.get('[data-testid="search-tasks"]').clear();
+  });
+
+  it("Testing Reset Progress", () => {
+    cy.wait(2000);
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts:138]: Avoid using wait, there are other solutions. If U need let's talk about it.

File: cypress/e2e/integration/MainPage.cy.ts [hunk 9]
```diff
@@ -0,0 +1,141 @@
+import { DateProps } from "interfaces/date_props.interface";
+
+describe("Renders Main page of app", () => {
+  let time: number = 0;
```
Comment (ssharabiappsforce) [inline cypress/e2e/integration/MainPage.cy.ts:4]: Remove the type, there is auto referring

File: cypress/e2e/integration/MainPage.cy.ts (hunk 3 above)
Comment (ohadsahar) [inline cypress/e2e/integration/MainPage.cy.ts:1]: Done

File: cypress/e2e/integration/MainPage.cy.ts (hunk 4 above)
Comment (ohadsahar) [reply cypress/e2e/integration/MainPage.cy.ts:23]: I'm using cypress 10.0.4, It's says there is no function like that.

File: core/components/TasksList/TasksList.tsx (hunk 2 above)
Comment (ohadsahar) [reply core/components/TasksList/TasksList.tsx:24]: Done

**Pull request closed without merging**